src/zarr/storage/_fsspec.py

```diff
@@ -261,6 +262,24 @@ async def delete(self, key: str) -> None:
         except self.allowed_exceptions:
             pass
 
+    async def delete_dir(self, prefix: str) -> None:
+        # docstring inherited
+        if not self.supports_deletes:
+            raise NotImplementedError('This method is only available for stores that support deletes.')
+        if not self.supports_listing:
+            raise NotImplementedError('This method is only available for stores that support directory listing.')
+        self._check_writable()
+
+        path_to_delete = _dereference_path(self.path, prefix)
+
+        if hasattr(self.fs, "_rm") and inspect.iscoroutinefunction(self.fs._rm):
```

Comment:
Add note: this is probably the same condition as `if self.fs.async_impl`

Reply:
note added